analyzer/codechecker_analyzer/cmd/analyze.py

```diff
@@ -655,7 +675,47 @@ def __get_skip_handler(args):
 
     if skip_file_content:
         LOG.debug_analyzer("Creating skiplist handler.")
-        return skiplist_handler.SkipListHandler(skip_file_content)
+        handler = skiplist_handler.SkipListHandler(skip_file_content)
+
+        analyze_headers = []
+
+        # Check whether the skip file contains a header file which is not
+        # skipped.
+        for skip_line in handler.skip_file_lines:
+            if skip_line[0] == '+' and \
+              skip_line.lower().endswith((".h", ".hh", ".hpp")):
+
+                norm_skip_path = os.path.normpath(skip_line[1:].strip())
+                rexpr = re.compile(
+                    fnmatch.translate(norm_skip_path + '*'))
+                analyze_headers.append((skip_line, rexpr))
+
+        # Get source files which depend on the previously collected header
+        # files and create a new skip list handler where we include these
+        # files in the beginning.
+        if analyze_headers:
+            dependencies = __get_source_dependencies(compile_commands)
+
+            analyze_header_deps = []
+            for f, deps in dependencies.items():
+                for _, rexpr in analyze_headers:
+                    if rexpr.match(f):
+                        analyze_header_deps.extend(["+" + d for d in deps])
+
+            if analyze_header_deps:
+                LOG.info("Your skip file contained some header files (%s). "
+                         "For this reason CodeChecker will include the "
+                         "following source files which depends on these "
+                         "header files:\n%s",
```

Comment:
```suggestion
                LOG.info("Your skip file contained some header files (%s) to be analyzed. "
                         "Analysis can not be executed on header files only. "
                         "For this reason CodeChecker will analyze the "
                         "following source files which include the header files "
                         ":\n%s",
```
I thinks a bit more explanation to the user would be nice here to better understand why those source files will be analyzed.